web/source/kmwosk.ts

```diff
@@ -3276,7 +3276,7 @@ if(!window['keyman']['initialized']) {
           if(n > 0 && shiftKey != null) {
             shiftKey['sp']=osk.buttonClasses['SHIFT-ON'];
             shiftKey['sk']=null;
-            shiftKey['text'] = osk.modifierSpecials[layers[n].id];
+            shiftKey['text'] = osk.modifierSpecials[layers[n].id] ? osk.modifierSpecials[layers[n].id] : "*Shift*";
```

Comment:
I think it would be a far better idea to simply extend the `osk.modifierSpecials` array to include non-chiral layer special key definitions.  That, or do both and use this as a last-case catch.

As this is, alternate layer shift keys for things like, say, `ctrl-alt`, will simply use the base `*Shift*` character at present as well rather than the more-consistent `*AltGr*` character that represents the CTRL+ALT shifting state.

Reply:
I've added labels for: `shift`, `ctrl`, `alt`, and `ctrl-alt` (`*AltGr*`)